ipvm-wg/homestar
# refactor: Upgrade svelvet

# Description

- Upgrade `svelvet` to v8(using new `Anchor`, `Edge` and `Node` components)
- Maintain Node positions, so running workflow one after workflow two will no long cause the Nodes to jump up to the first row

## Link to issue

https://github.com/ipvm-wg/homestar/issues/337

## Type of change

- [x] Refactor (non-breaking change that updates existing functionality)

## Test plan (required)

1. Run `cargo run -- start -c config/settings.toml` from `homestar/examples/websocket-relay`
2. Run `npm run dev` from `homestar/examples/websocket-relay/relay-app`
3. Visit http://localhost:5173/
4. Run workflow one and expect the Nodes and edges to appear mostly as they did before(black edges with arrows and labels describing the associated tasks)
5. Run workflow two and expect the grayscale Node to appear in the bottom row with an edge connected to its north side
6. Refresh the page
7. Run workflow two and expect the Nodes, Edges and labels to appear correctly in the bottom row
8. Run workflow one and expect the Nodes that have already been executed to stay in the bottom row, but also have the blurred Node rendered in the top row with an Edge connected to it

## Screenshots/Screencaps

https://www.loom.com/share/acc35d252e8b4d5a94a6f0d53aecfa91?sid=54cdb36b-1837-4082-b759-f8e80f36503e
